### PR TITLE
Adding Arguments to some get functions ...

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

Adding missing arguments to getter functions

**Why**:

Missing arguments lead to missing opportunities ...

**How**:

Using the arguments of the multiple-getter functions and integrated them to the single-getters

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: tests/protocols/gmpv7/testcmds/test_get_credential.py [hunk 1]
```diff
@@ -77,6 +77,32 @@ def test_get_credential_with_invalid_credential_format(self):
         with self.assertRaises(InvalidArgumentType):
             self.gmp.get_credential('id', credential_format='foo')
 
+    def test_get_credentials_with_scanners(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv7/testcmds/test_get_credential.py:80]:
```suggestion
    def test_get_credential_with_scanners(self):
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 2]
```diff
@@ -2777,7 +2777,7 @@ def get_alerts(
 
         return self._send_xml_command(cmd)
 
-    def get_alert(self, alert_id: str) -> Any:
+    def get_alert(self, alert_id: str, tasks: Optional[bool] = None) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:2780]: Optional arguments should always be required to be passed as keywords.
```suggestion
    def get_alert(self, alert_id: str, *, tasks: Optional[bool] = None) -> Any:
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 3]
```diff
@@ -2895,13 +2901,18 @@ def get_credentials(
     def get_credential(
         self,
         credential_id: str,
+        scanners: Optional[bool] = None,
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:2904]:
```suggestion
        *,
        scanners: Optional[bool] = None,
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 4]
```diff
@@ -2975,7 +2993,7 @@ def get_configs(
 
         return self._send_xml_command(cmd)
 
-    def get_config(self, config_id: str) -> Any:
+    def get_config(self, config_id: str,) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:2996]: Fix for python 3.5
```suggestion
    def get_config(self, config_id: str) -> Any:
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 5]
```diff
@@ -3061,22 +3080,28 @@ def get_filters(
 
         return self._send_xml_command(cmd)
 
-    def get_filter(self, filter_id: str) -> Any:
+    def get_filter(self, filter_id: str, alerts: Optional[bool] = None) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3083]:
```suggestion
    def get_filter(self, filter_id: str, *, alerts: Optional[bool] = None) -> Any:
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 6]
```diff
@@ -3694,22 +3748,25 @@ def get_report_formats(
 
         return self._send_xml_command(cmd)
 
-    def get_report_format(self, report_format_id: str) -> Any:
+    def get_report_format(self, report_format_id: str,) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3751]:
```suggestion
    def get_report_format(self, report_format_id: str) -> Any:
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 7]
```diff
@@ -3758,7 +3815,7 @@ def get_results(
 
         return self._send_xml_command(cmd)
 
-    def get_result(self, result_id: str) -> Any:
+    def get_result(self, result_id: str,) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3818]:
```suggestion
    def get_result(self, result_id: str) -> Any:
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 8]
```diff
@@ -3907,21 +3968,28 @@ def get_schedules(
 
         return self._send_xml_command(cmd)
 
-    def get_schedule(self, schedule_id: str) -> Any:
+    def get_schedule(
+        self, schedule_id: str, tasks: Optional[bool] = None
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3972]:
```suggestion
        self, schedule_id: str, *, tasks: Optional[bool] = None
```

File: gvm/protocols/gmpv7/gmpv7.py [hunk 9]
```diff
@@ -4090,22 +4160,28 @@ def get_targets(
 
         return self._send_xml_command(cmd)
 
-    def get_target(self, target_id: str) -> Any:
+    def get_target(self, target_id: str, tasks: Optional[bool] = None) -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:4163]:
```suggestion
    def get_target(self, target_id: str, *, tasks: Optional[bool] = None) -> Any:
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: gvm/protocols/gmpv7/gmpv7.py [hunk 10]
```diff
@@ -3179,10 +3207,15 @@ def get_info(self, info_id: str, info_type: InfoType) -> Any:
             info_id: UUID of an existing secinfo
             info_type: Type must be either CERT_BUND_ADV, CPE, CVE,
                 DFN_CERT_ADV, OVALDEF, NVT or ALLINFO
+            name: Name or identifier of the requested information
+            details: Whether to include information about references to this
+                information
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3212]: These parameters are not implemented right?

File: gvm/protocols/gmpv7/gmpv7.py [hunk 11]
```diff
@@ -3395,16 +3429,19 @@ def get_override(self, override_id: str) -> Any:
 
         Arguments:
             override_id: UUID of an existing override
+            details: Whether to include full details
+            result: Whether to include results using the override
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3433]: Same. I don't see the implementation of these arguments.

File: gvm/protocols/gmpv7/gmpv7.py [hunk 12]
```diff
@@ -3496,16 +3534,19 @@ def get_port_list(self, port_list_id: str):
 
         Arguments:
             port_list_id: UUID of an existing port list
+            details: Whether to include full port list details
+            targets: Whether to include targets using this port list
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3538]: Same. I don't see the implementation of these arguments.

File: gvm/protocols/gmpv7/gmpv7.py [hunk 13]
```diff
@@ -3860,16 +3919,19 @@ def get_scanner(self, scanner_id: str) -> Any:
 
         Arguments:
             scanner_id: UUID of an existing scanner
+            details:  Whether to include extra details like tasks using this
+                scanner
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:3923]: The implementation for details is missing

Comment (y0urself) [reply gvm/protocols/gmpv7/gmpv7.py:3923]: I first added these parameters, but then noticed:
```
# for single entity always request all details
```
So these parameters are not necessary ... (I hope ...)

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**